server/middlewares/checkUser.js

```diff
@@ -0,0 +1,21 @@
+import passport from '../services/passport'
+
+// This for setting the user value when the festival is free for checing if user is owner later.
+
+function checkJWT(req, res, next) {
+  passport.authenticate('jwt', { session: false }, (err, user) => {
+    if (err) {
+      return next(err)
+    }
+    if (!user) {
+      req.user = {}
+      return next()
+    }
+    req.user = user
+    return next()
+  })(req, res, next)
+}
```

Comment:
Maybe like this:

```
export function getUserWhenAuthenticated(req, res, next) {
  passport.authenticate('jwt', { session: false }, (err, user) => {
    if (err) {
      return next(err)
    }

    if (user) {
      req.user = user
    }

    next()
  })(req, res, next)
}
```

* Maybe rename `checkUser` to `getUserWhenAuthenticated` or similar. Its fine as it is, but when its about permissions (or not as in this case) it might be good to be more explicit
* We might need to check if anywhere there is a need where `req.user` is set to make sure nothing breaks. I think its better to not set it at all than giving it an empty object (as this is in some cases "technically" set as well).

Reply:
I made those changes here https://github.com/adzialocha/hoffnung3000/pull/82/commits/01cfa050c335c242fd1a98e95eaffe24740627d7 but it seems that `user` does need to be set as an empty object for the controller to pass ok, so I just added that.